access_log/access_log_record.go

```diff
@@ -94,3 +100,14 @@ func (r *AccessLogRecord) LogMessage() string {
 	recordBuffer := r.makeRecord()
 	return recordBuffer.String()
 }
+
+func (r *AccessLogRecord) ExtraHeaders() string {
+	extraHeaders := []string{}
+	for _, header := range r.ExtraHeadersToLog {
+		// X-Something-Cool -> x_something_cool
+		formatted_header_name := strings.Replace(strings.ToLower(header), "-", "_", -1)
+		headerString := fmt.Sprintf("%s:%s", formatted_header_name, r.FormatRequestHeader(header))
```

Comment:
I think we should double quote the header values when we format to ensure we handle lists and so parsing of log files should be predictable (key:"value"  or key:"value1, value2")

For example: If we wanted to log the X-Forwarded-Proto and X-Forwarded-For headers
```
X-Forwarded-Proto: https, http
X-Forwarded-For: 10.0.0.1, 10.0.0.2
```

We would want the access log elements to look like this:

```
x_forwarded_proto:"https, http" x_forwarded_for:"10.0.0.1, 10.0.0.2"
```

vs. 

```
x_forwarded_proto:https, http x_forwarded_for:10.0.0.1, 10.0.0.2
```

Cheers